presto-main/src/main/java/com/facebook/presto/sql/planner/EffectivePredicateExtractor.java

```diff
@@ -278,9 +264,12 @@ private Expression deriveCommonPredicates(PlanNode node, Function<Integer, Colle
         for (int i = 0; i < node.getSources().size(); i++) {
             Expression underlyingPredicate = node.getSources().get(i).accept(this, null);
 
-            Iterable<Expression> equalities = FluentIterable.from(mapping.apply(i))
-                    .filter(not(symbolMatchesExpression()))
-                    .transform(entryToEquality());
+            List<Expression> equalities = mapping.apply(i).stream()
```

Comment:
This is much harder to read. Leave the predicate methods (but convert them to JDK Predicate)